### PR TITLE
build: update to ember 3.24

### PR DESCRIPTION
**Pull request by GreatWizard**
### Update to ember 3.24 (LTS) (#112)

Updates all dependencies/configuration to please ember 3.24

### CODE REVIEW

File: .template-lintrc.js [hunk 1]
```diff
@@ -1,54 +1,5 @@
-'use strict'
+'use strict';
```
Comment (MrChocolatine) [inline .template-lintrc.js:1]: The overwritten configuration was the common one across all our add-ons, we should keep it.

Comment (GreatWizard) [reply .template-lintrc.js]: :+1: I revert it

**Review by ndekeister-us — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -14,50 +14,59 @@
...
     "ember-disable-prototype-extensions": "^1.1.3",
     "ember-export-application-global": "^2.0.1",
-    "ember-load-initializers": "^2.1.1",
+    "ember-load-initializers": "^2.1.2",
     "ember-maybe-import-regenerator": "^0.1.6",
-    "ember-qunit": "^4.6.0",
-    "ember-resolver": "^8.0.0",
-    "ember-sinon-qunit": "^5.0.0",
-    "ember-source": "~3.27.2",
+    "ember-page-title": "^6.0.3",
```
Comment (ndekeister-us) [inline package.json:52]: @GreatWizard you could remove this package as our dummy app is empty so we don't really use it / need it (+ remove `{{page-title "Dummy"}}` in `tests/dummy/app/templates/application.hbs`)

File: package.json [hunk 3]
```diff
@@ -14,49 +14,57 @@
...
+    "test": "npm-run-all lint test:*",
     "test:ember": "ember test",
     "test:ember-compatibility": "ember try:each"
   },
   "dependencies": {
     "ember-classic-decorator": "^2.0.0",
-    "ember-cli-babel": "^7.21.0",
-    "ember-cli-htmlbars": "^5.7.1",
+    "ember-cli-babel": "^7.23.0",
+    "ember-cli-htmlbars": "^5.3.1",
```
Comment (MrChocolatine) [inline package.json:31]: We are making a downgrade, we can undo the change for this line `ember-cli-htmlbars`.

File: package.json [hunk 4]
```diff
@@ -14,49 +14,57 @@
...
-    "ember-load-initializers": "^2.1.1",
+    "ember-load-initializers": "^2.1.2",
     "ember-maybe-import-regenerator": "^0.1.6",
-    "ember-qunit": "^4.6.0",
-    "ember-resolver": "^8.0.0",
+    "ember-qunit": "^5.1.1",
+    "ember-resolver": "^8.0.2",
     "ember-sinon-qunit": "^5.0.0",
-    "ember-source": "~3.27.2",
+    "ember-source": "~3.24.0",
```
Comment (MrChocolatine) [inline package.json:54]: We are making a downgrade, we can undo the change for this line `ember-source`.

File: package.json [hunk 5]
```diff
@@ -14,49 +14,57 @@
...
     "ember-source-channel-url": "^3.0.0",
-    "ember-template-lint": "^3.3.1",
+    "ember-template-lint": "^2.15.0",
```
Comment (MrChocolatine) [inline package.json:56]: We are making a downgrade, we can undo the change for this line `ember-template-lint`.

File: package.json [hunk 6]
```diff
@@ -14,49 +14,57 @@
...
     "ember-try": "^1.4.0",
-    "eslint": "^7.25.0",
-    "eslint-plugin-ember": "^10.4.1",
+    "eslint": "^7.17.0",
```
Comment (MrChocolatine) [inline package.json:58]: We are making a downgrade, we can undo the change for this line `eslint`.

File: package.json [hunk 7]
```diff
@@ -14,49 +14,57 @@
...
+    "eslint-config-prettier": "^7.1.0",
+    "eslint-plugin-ember": "^10.1.1",
```
Comment (MrChocolatine) [inline package.json:60]: We are making a downgrade, we can undo the change for this line `eslint-plugin-ember`.

**Review by MrChocolatine — CHANGES_REQUESTED** [review]

**Review by MrChocolatine — APPROVED** [review]
